plugin/db/mysql/mysql.go

```diff
@@ -913,21 +914,30 @@ func (driver *Driver) Dump(ctx context.Context, database string, out io.Writer,
 }
 
 // Restore restores a database.
-func (driver *Driver) Restore(ctx context.Context, sc *bufio.Scanner) (err error) {
+func (driver *Driver) Restore(ctx context.Context, sc *bufio.Scanner, config db.RestoreConfig) (err error) {
 	txn, err := driver.db.BeginTx(ctx, nil)
 	if err != nil {
 		return err
 	}
 	defer txn.Rollback()
 
-	f := func(stmt string) error {
+	fnExecuteStmt := func(stmt string) error {
 		if _, err := txn.Exec(stmt); err != nil {
 			return err
 		}
 		return nil
 	}
 
-	if err := util.ApplyMultiStatements(sc, f); err != nil {
+	// restore to ghost tables, need to replace table names
```

Comment:
I would expand the comment a bit. "need to replace table names" is the key step. IIUC, we are generating the new ghost table using the name based on the existing table name.